tests/backup/backup_rbac_test.go

```diff
@@ -499,3 +499,433 @@ var _ = Describe("{VerifyRBACforInfraAdmin}", func() {
+			log.InfoD(fmt.Sprintf("Creating a user with %s role", appAdminRole))
+			appAdminUser = createUsers(1)[0]
+			err := backup.AddRoleToUser(appAdminUser, appAdminRole, fmt.Sprintf("Adding %v role to %s", appAdminRole, appAdminUser))
+			log.FailOnError(err, "failed to add role %s to the user %s", appAdminRole, appAdminUser)
+		})
+
+		Step("Verify App-Admin User doesnt have permission to create cloud credential", func() {
+			log.InfoD("Verify App-Admin User doesnt have permission to create cloud credential")
+			nonAdminCtx, err := backup.GetNonAdminCtx(appAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appAdminUser)
+			for _, provider := range providers {
+				if provider != drivers.ProviderNfs {
+					cloudCredentialName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+					cloudCredentialUID := uuid.New()
+					err = CreateCloudCredential(provider, cloudCredentialName, cloudCredentialUID, orgID, nonAdminCtx)
+					dash.VerifyFatal(strings.Contains(err.Error(), "PermissionDenied desc = Access denied for [Resource: cloudcredential]"), true, fmt.Sprintf("Verifying creation failure of cloud credential %s using provider %s for the user", cloudCredentialName, provider))
+				}
+			}
+		})
+
+		Step("Verify App-Admin User doesnt have permission to create role", func() {
+			log.InfoD("Verify App-Admin User doesnt have permission to create role")
+			nonAdminCtx, err := backup.GetNonAdminCtx(appAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appAdminUser)
+			customRoleName = backup.PxBackupRole(fmt.Sprintf("custom-user-role-%s", RandomString(4)))
+			services := []RoleServices{SchedulePolicy, Rules, Cloudcredential, BackupLocation, Role}
+			apis := []RoleApis{All}
+			err = CreateRole(customRoleName, services, apis, nonAdminCtx)
+			dash.VerifyFatal(strings.Contains(err.Error(), "PermissionDenied desc = Access denied for [Resource: role]"), true, fmt.Sprintf("Verifying app-user doesnt have permission for creating role [%s]", customRoleName))
+		})
+
+		Step(fmt.Sprintf("Adding Credentials and BackupLocation from px-admin user and making it public"), func() {
+			log.InfoD(fmt.Sprintf("Adding Credentials and BackupLocation from px-admin user and making it public"))
+			for _, provider := range providers {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-admin ctx")
+				adminCloudCredUID = uuid.New()
+				adminCredName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				err = CreateCloudCredential(provider, adminCredName, adminCloudCredUID, orgID, ctx)
+				log.FailOnError(err, "Failed to create cloud credential - %s", err)
+				if provider != drivers.ProviderNfs {
+					err = UpdateCloudCredentialOwnership(adminCredName, adminCloudCredUID, nil, nil, Invalid, Read, ctx, orgID)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying public ownership update for cloud credential %s ", adminCredName))
```

Comment:
Is it ok if we move 'if provider != drivers.ProviderNfs' before CreateCloudCredential?